src/util/routerinfo.cc

```diff
@@ -118,16 +121,33 @@ bool RouterInfoCommand::Impl(
                                                    core::RouterInfo::MaxPort)
                                              : vm["port"].as<int>();
 
+          // Getting the ip address list out of the 'host' param.
+          boost::char_separator<char> token_separator{","};
+          boost::tokenizer<boost::char_separator<char>> host_tokens{
+              host, token_separator};
+          // prepare the host list.
+          core::RouterInfo::AddressInfoList hosts;
+          for (const auto& host_token : host_tokens)
+            {
+              core::RouterInfo::AddressInfo address;
+              address.host = host_token;
+              address.port = port;
+              address.has_transport = std::make_pair(true, true);
+              hosts.push_back(std::move(address));
+            }
+          // ipv4 and ipv6 allowed only.
+          if (hosts.size() > 2)
+            {
+              LOG(error) << "routerinfo: Too many hosts in the input parameter";
+              return false;
+            }
+
           std::uint8_t caps(core::RouterInfo::Cap::Reachable);
           // Generate private key
           core::PrivateKeys keys = core::PrivateKeys::CreateRandomKeys(
               core::DEFAULT_ROUTER_SIGNING_KEY_TYPE);
           // Create router info
-          core::RouterInfo routerInfo(
-              keys,
-              std::make_pair(host, port),
-              std::make_pair(true, true),
```

Comment:
This reminds me that there is a bug that I've been meaning to address. As found on testnet, when disabling transports, they are not appropriately disabled.